packages/cli/test/unit/services/user.service.test.ts

```diff
@@ -7,30 +7,79 @@ import { UserRepository } from '@db/repositories/user.repository';
 import { UserService } from '@/services/user.service';
 import { mockInstance } from '../../shared/mocking';
 import { RoleService } from '@/services/role.service';
+import { v4 as uuid } from 'uuid';
 
 describe('UserService', () => {
 	config.set('userManagement.jwtSecret', 'random-secret');
 
 	mockInstance(Logger);
-	const repository = mockInstance(UserRepository);
 	mockInstance(RoleService);
-	const service = Container.get(UserService);
-	const testUser = Object.assign(new User(), {
-		id: '1234',
+
+	const userRepository = mockInstance(UserRepository);
+	const userService = Container.get(UserService);
+
+	const commonMockUser = Object.assign(new User(), {
+		id: uuid(),
 		password: 'passwordHash',
-		mfaEnabled: false,
 	});
 
-	beforeEach(() => {
-		jest.resetAllMocks();
+	describe('toPublic', () => {
+		it('should remove sensitive properties', async () => {
+			const mockUser = Object.assign(new User(), {
+				id: uuid(),
+				password: 'passwordHash',
+				mfaEnabled: false,
+				mfaSecret: 'test',
+				mfaRecoveryCodes: ['test'],
+				updatedAt: new Date(),
+				authIdentities: [],
+			});
+
+			type MaybeSensitiveProperties = Partial<
+				Pick<User, 'password' | 'mfaSecret' | 'mfaRecoveryCodes' | 'updatedAt' | 'authIdentities'>
+			>;
+
+			// to prevent typechecking from blocking assertions
+			const publicUser: MaybeSensitiveProperties = await userService.toPublic(mockUser);
+
+			expect(publicUser.password).toBeUndefined();
+			expect(publicUser.mfaSecret).toBeUndefined();
+			expect(publicUser.mfaRecoveryCodes).toBeUndefined();
+			expect(publicUser.updatedAt).toBeUndefined();
+			expect(publicUser.authIdentities).toBeUndefined();
+		});
+
+		it('should add scopes if requested', async () => {
+			const scopeless = await userService.toPublic(commonMockUser, { withScopes: false });
+
+			const scoped = await userService.toPublic(commonMockUser, { withScopes: true });
+
+			expect(Array.isArray(scopeless.globalScopes)).toBe(false);
+			expect(Array.isArray(scoped.globalScopes)).toBe(true);
+		});
+
+		it('should add invite URL if requested', async () => {
+			const mockUser = Object.assign(new User(), { id: uuid(), isPending: true });
+
+			const withUrl = await userService.toPublic(mockUser, { withInviteUrl: true });
+			const withoutUrl = await userService.toPublic(mockUser, { withInviteUrl: false });
+
+			expect(typeof withUrl.inviteAcceptUrl === 'string').toBe(true);
```

Comment:
We should also assert that the URL is valid. For example should include the ownerId and the memberId. This is currently broken as mentioned [here](https://linear.app/n8n/issue/ADO-1494/on-user-list-copy-password-reset-link-and-copy-invite-link-are-broken). The invitation URL returns the ownerid for invitee and inviter

Reply:
Thanks! I noticed this - should we make the fix in that story separate from this PR to expand tests?